api/v1/helpers.go

```diff
@@ -2211,12 +2248,22 @@ func (v *VerticaDB) GetClientServerTLSSecret() string {
 	return v.Spec.ClientServerTLS.Secret
 }
 
-// Check if TLS not enabled, DB not initialized, or rotate has failed (and rollback is not in progress).
-// In these cases, we skip TLS Update
-func (v *VerticaDB) ShouldSkipTLSUpdateReconcile() bool {
-	return !v.IsSetForTLS() ||
+// ShouldSkipTLSUpdateReconcile will check if TLS Update can be skipped for this config.
+// Possible reasons are:
+// 1) If TLS not enabled
+// 2) DB not initialized
+// 3) Rotate has failed (and rollback is not in progress)
+// 4) TLS auth is disabled for this config
+func (v *VerticaDB) ShouldSkipTLSUpdateReconcile(tlsConfigName string, isRollback bool) bool {
```

Comment:
You could simplify this by having a helper function and then have one version for each tls config.